src/ensembl/src/content/app/entity-viewer/gene-view/shared/views-helpers.ts

```diff
@@ -0,0 +1,51 @@
+/**
+ * See the NOTICE file distributed with this work for additional information
+ * regarding copyright ownership.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  GeneFunctionTabName,
+  GeneRelationshipsTabName,
+  GeneViewTabName
+} from 'src/content/app/entity-viewer/state/gene-view/entityViewerGeneViewState';
+
+export const geneFunctionViews: { [key: string]: string } = {
+  protein: GeneFunctionTabName.PROTEINS,
+  variants: GeneFunctionTabName.VARIANTS,
+  phenotypes: GeneFunctionTabName.PHENOTYPES,
+  gene_expression: GeneFunctionTabName.GENE_EXPRESSION,
+  gene_ontology: GeneFunctionTabName.GENE_ONTOLOGY,
+  gene_pathways: GeneFunctionTabName.GENE_PATHWAYS
+};
+
+export const geneRelationshipsViews: { [key: string]: string } = {
+  orthologues: GeneRelationshipsTabName.ORTHOLOGUES,
+  gene_families: GeneRelationshipsTabName.GENE_FAMILIES,
+  gene_panels: GeneRelationshipsTabName.GENE_PANELS
+};
+
+export const getViewModeName = (
+  geneViewTab: GeneViewTabName,
+  childTab?: string
```

Comment:
why is it nullable? according to the code below, it's used in the `find` method

Reply:
It is optional in this case as it is a param. When using this function to get the transcript view there will be no child tab, hence why it is optional.